pkg/operation/botanist/component/shared/fluent_operator_resources.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package shared
+
+import (
+	"github.com/Masterminds/semver"
+	fluentbitv1alpha2 "github.com/fluent/fluent-operator/apis/fluentbit/v1alpha2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/gardener/gardener/pkg/operation/botanist/component"
+	"github.com/gardener/gardener/pkg/operation/botanist/component/logging/fluentoperator"
+	"github.com/gardener/gardener/pkg/utils/images"
+	"github.com/gardener/gardener/pkg/utils/imagevector"
+)
+
+// NewFluentOperatorResources instantiates a new `fluent-operator-custom-resources` component.
+func NewFluentOperatorResources(
+	c client.Client,
+	gardenNamespaceName string,
+	runtimeVersion *semver.Version,
+	imageVector imagevector.ImageVector,
+	enabled bool,
+	priorityClassName string,
+	additionalInputs []*fluentbitv1alpha2.ClusterInput,
+	additionalFilters []*fluentbitv1alpha2.ClusterFilter,
+	additionalParsers []*fluentbitv1alpha2.ClusterParser,
+) (
+	deployer component.DeployWaiter,
+	err error,
+) {
+	fluentBitImage, err := imageVector.FindImage(images.ImageNameFluentBit)
+	if err != nil {
+		return nil, err
+	}
+
+	fluentBitInitImage, err := imageVector.FindImage(images.ImageNameFluentBitPluginInstaller)
+	if err != nil {
+		return nil, err
+	}
+
+	deployer = fluentoperator.NewCustomResources(
+		c,
+		gardenNamespaceName,
+		fluentoperator.CustomResourcesValues{
+			FluentBitImage:         fluentBitImage.String(),
+			FluentBitInitImage:     fluentBitInitImage.String(),
+			FluentBitPriorityClass: priorityClassName,
+		},
+		additionalInputs,
+		additionalFilters,
+		additionalParsers,
+	)
+
+	if !enabled {
+		deployer = component.OpDestroyWithoutWait(deployer)
+	}
```

Comment:
I disabled the logging locally and the `fluent-operator-resources` MR was not able to be deleted with:
```
  - lastTransitionTime: "2023-03-29T13:24:37Z"
    lastUpdateTime: "2023-03-29T13:24:37Z"
    message: 'Could not clean all old resources: 1 error occurred: deletion of old
      resource "fluentbit.fluent.io/v1alpha2/FluentBit/garden/fluent-bit" is still
      pending'
    reason: DeletionPending
    status: Progressing
    type: ResourcesApplied
```

The fluentbit resource stil had a finalizer
```yaml
    finalizers:
    - fluentbit.fluent.io
```

while the fluent-operator was already deleted by the Seed reconciliation.

Reply:
Yes, we should wait the fluent-bit CR to be destroyed before destroying the fluent-operator.

Reply:
I don't think this comment is resolved. I disable the logging for a Seed and the `fluent-operator-resources` MR leaked as described above.
I think in https://github.com/gardener/gardener/blob/c3c1a922d336982486adf85c6a9c4636abfb1367/pkg/gardenlet/controller/seed/seed/reconciler_reconcile.go#L1013-L1020 you have to put dependency between the tasks. `Deploying Fluent Operator` should be executed only after `Deploying Fluent Operator resources and wait until they become ready`.